UPGRADE.rst

```diff
@@ -55,50 +55,140 @@ Upgrading to v1.4.0
+
+The ``account_threepid_delegates`` dictionary defines whether the homeserver
+should delegate an external server (typically an `identity server
+<https://matrix.org/docs/spec/identity_service/r0.2.1>`_) to handle sending
+password reset or registration messages via email and SMS.
+
+So to delegate email sending set ``account_threepid_delegates.email`` to a base
+URL of an identity server in your homeserver.yaml. Note that
+``account_threepid_delegates.email`` replaces the deprecated
+``email.trust_identity_server_for_password_resets``.
+
+.. code:: yaml
+
+   account_threepid_delegates:
+       email: https://example.com     # Delegate email sending to example.com
 
 If ``email.trust_identity_server_for_password_resets`` is set to ``true``, and
 ``account_threepid_delegates.email`` is not set, then the first entry in
-``trusted_third_party_id_servers`` will be used as the account threepid delegate for email.
-This is to ensure compatibility with existing Synapse installs that set up external server
-handling for these tasks before v1.4.0. If ``email.trust_identity_server_for_password_resets``
-is ``true`` and no trusted identity server domains are configured, Synapse will throw an error.
+``trusted_third_party_id_servers`` will be used as the account threepid
+delegate for email. This is to ensure compatibility with existing Synapse
+installs that set up external server handling for these tasks before v1.4.0. If
+``email.trust_identity_server_for_password_resets`` is ``true`` and no trusted
+identity server domains are configured, Synapse will throw an error.
 
-If ``email.trust_identity_server_for_password_resets`` is ``false`` or absent and a threepid
-type in ``account_threepid_delegates`` is not set to a domain, then Synapse will attempt to
-send password reset and registration messages for that type.
+If ``email.trust_identity_server_for_password_resets`` is ``false`` or absent
+and a threepid type in ``account_threepid_delegates`` is not set to a domain,
+then Synapse will attempt to send password reset and registration messages for
+that type.
 
 Email templates
 ---------------
 
-If you have configured a custom template directory with the ``email.template_dir`` option, be
-aware that there are new templates regarding registration. ``registration.html`` and
-``registration.txt`` have been added and contain the content that is sent to a client upon
-registering via an email address.
+If you have configured a custom template directory with the
+``email.template_dir`` option, be aware that there are new templates regarding
+registration and threepid management.
+
+``registration.html`` and ``registration.txt`` contain the content that is sent
+to a client upon registering via an email address.
+
+``registration_success.html`` and ``registration_failure.html`` are templates
+that will be shown to the user when they click the link in their registration
+email , either showing them a success or failure page (assuming a redirect URL
```

Comment:
A redirect URL is not a thing that gets configured. It's a bit awkward to describe:
 
```suggestion
email, either showing them a success page (in the case that no onward redirect was specified by the client), or a failure page.
```